Trustroots/trustroots
# Admin search users

#### Proposed Changes

* First /admin stuff: `/admin/search-users`, using the `users?search=query` endpoint
* Adds `requireRole` to angular stuff

#### Testing Instructions

* add `admin` to your test user's `roles`
* go to http://localhost:3000/admin

Relates to #857, #530

Next steps:
- [x] create admin API endpoint on server that searches on emails, and returns emails
- [x] `/api/admin/users` should only be accessible by admin users - ensure with a test
- [x] use onSubmit instead of onChange or use debounce
- [ ] show recent down-voted threads
- [ ] show how new people heard about trustroots #879

(still nice to already merge this before next deployment to prod)
